test/suites/test_interval.py

```diff
@@ -0,0 +1,206 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import print_function
+
+import sys
+import unittest
+import msgpack
+import warnings
+import tarantool
+import pandas
+import pytz
+
+from tarantool.msgpack_ext.packer import default as packer_default
+from tarantool.msgpack_ext.unpacker import ext_hook as unpacker_ext_hook
+
+from .lib.tarantool_server import TarantoolServer
+from .lib.skip import skip_or_run_datetime_test
+from tarantool.error import MsgpackError
+
+class TestSuite_Interval(unittest.TestCase):
+    @classmethod
+    def setUpClass(self):
+        print(' INTERVAL EXT TYPE '.center(70, '='), file=sys.stderr)
+        print('-' * 70, file=sys.stderr)
+        self.srv = TarantoolServer()
+        self.srv.script = 'test/suites/box.lua'
+        self.srv.start()
+
+        self.adm = self.srv.admin
+        self.adm(r"""
+            _, datetime = pcall(require, 'datetime')
+
+            box.schema.space.create('test')
+            box.space['test']:create_index('primary', {
+                type = 'tree',
+                parts = {1, 'string'},
+                unique = true})
+
+            box.schema.user.create('test', {password = 'test', if_not_exists = true})
+            box.schema.user.grant('test', 'read,write,execute', 'universe')
+        """)
+
+        self.con = tarantool.Connection(self.srv.host, self.srv.args['primary'],
+                                        user='test', password='test')
+
+    def setUp(self):
+        # prevent a remote tarantool from clean our session
+        if self.srv.is_started():
+            self.srv.touch_lock()
+
+        self.adm("box.space['test']:truncate()")
+
+
+    cases = {
+        'year': {
+            'python': tarantool.Interval(year=1),
+            'msgpack': (b'\x02\x00\x01\x08\x01'),
+            'tarantool': r"datetime.interval.new({year=1})",
+        },
```

Comment:
There is no test cases for:

tarantool.Interval(data=foobar)

Reply:
Added

Reply:
Unknown interval field and re-raising Adjust ValueError already tested in `test_unknown_field_decode` and `test_unknown_adjust_decode`. I don't think it's worth to test them one more time but with direct `data=bytes''` call.

Reply:
Regarding the first one, I've added `test_Interval_bytes_init_ignore_other_fields`

Reply:
However, there is a difference between the unit test for the constructor and for the decode function from msgpack.

Reply:
Well, ok, I've added these tests + test on `data` is `bytes`